tap_agent/src/config.rs

```diff
@@ -0,0 +1,320 @@
+// Copyright 2023-, GraphOps and Semiotic Labs.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::path::PathBuf;
+
+use alloy_primitives::Address;
+use clap::{command, Args, Parser, ValueEnum};
+use dotenvy::dotenv;
+use serde::{Deserialize, Serialize};
+use toolshed::thegraph::DeploymentId;
+use tracing::subscriber::{set_global_default, SetGlobalDefaultError};
+use tracing_subscriber::{EnvFilter, FmtSubscriber};
+
+#[derive(Clone, Debug, Parser, Serialize, Deserialize, Default)]
+#[clap(
+    name = "indexer-tap-agent",
+    about = "Agent that manages Timeline Aggregation Protocol (TAP) receipts as well \
+    as Receipt Aggregate Voucher (RAV) requests for an indexer."
+)]
+#[command(author, version, about, long_about = None, arg_required_else_help = true)]
+pub struct Cli {
```

Comment:
Some of these are duplicates of `indexer-service CLI`, would it make sense to refactor the duplicates to `indexer-common`?

Reply:
Yup! Not sure there is an elegant way to do that though.

Reply:
Actually in k8s, the preferred way of passing secrets is using env vars. Unless you are willing to integrate your software directly with the k8s or your cloud's secrets API.
My favorite config implementation is that of `graph-node`. It uses TOML, but also does something like `envsubst` over the file before loading it.

Reply:
> https://www.cloudtruth.com/blog/the-pitfalls-of-using-environment-variables-for-config-and-secrets suggests that environment variables for secrets is insecure.

I'm not sure this really applies to containers though. This is still one of the recommended ways in k8s for ex: https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/